dnf/conf/config.py

```diff
@@ -175,6 +175,22 @@ def dump(self):
 
         return '\n'.join(output) + '\n'
 
+    def set_or_append_opt_value(self, name, value_string, priority=PRIO_RUNTIME):
+        # :api
+        """For standard options, sets the value of the option if the priority is equal to or higher
+           than the current priority.
+           For "append" options, appends the values parsed from value_string to the current list of values. If the first
+           parsed element of the list of values is empty and the priority is equal to or higher than the current
+           priority, the current list is replaced with the new values.
+           If the priority is higher than the current priority, the current priority is increased to the priority.
```

Comment:
May I ask you to add information about exceptions?

Reply:
Yes.